releases/enterprise.go

```diff
@@ -0,0 +1,29 @@
+package releases
+
+import (
+	"fmt"
+)
+
+type EnterpriseOptions struct {
+	Enterprise bool
+	Meta       string // optional; may be "hsm", "fips1402", "hsm.fips1402", etc.
+	LicenseDir string // required when Enterprise is true
+}
+
+func (eo *EnterpriseOptions) requiredMetadata() string {
+	metadata := ""
+	if eo.Enterprise {
+		metadata += "ent"
+	}
+	if eo.Meta != "" {
+		metadata += "." + eo.Meta
+	}
+	return metadata
+}
+
+func (eo *EnterpriseOptions) validate() error {
+	if eo.Enterprise && eo.LicenseDir == "" {
+		return fmt.Errorf("license dir must be provided when requesting enterprise versions")
+	}
```

Comment:
```suggestion
	if eo.LicenseDir == "" {
		return fmt.Errorf("LicenseDir must be provided when requesting enterprise versions")
	}
```

We could use `EnterpriseOptions` as a pointer - that allows us to differentiate between set/unset (as pointer can default to `nil`, unlike struct). It looks like you already declared all methods as methods on pointers anyway.

Reply:
Boolean removed via 2eaee33 and error message adjusted via ca8e21b